src/pybel/io/mongodb.py

```diff
@@ -0,0 +1,203 @@
+    db_name: Optional[str] = None,
+    collection_name: Optional[str] = None,
+    client: Optional[MongoClient] = None
+) -> Collection:
+    """Export the given BELGraph to a MongoDB.
+
+    In order to use this function, MongoDB must already be running locally.
+
+    :param graph: The graph to be exported
+    :param db_name: An optional name of the MongoDB to which the graph should be exported
+    :param collection_name: An optional name of the collection within the MongoDB where the graph will be stored.
+    :param client: An optional MongoClient object to use instead of a default
+    :return: the collection that now stores the graph
+    """
+    if client is None:
+        client = MongoClient()
+    if db_name is None:
+        db_name = DB_NAME
+    # Access (or create) the specified database
+    db = client[db_name]
+    # If no collection_name was passed
+    if collection_name is None:
+        # Attempt to create collections until a unique name is found
+        collection_name = _default_collection_name(db)
+    # Access (or create) the specified collection
+    collection = db[collection_name]
+    # If a collection with the same name already exists, drop its contents
+    collection.drop()
+
+    # Store all the nodes/edges in a list to be bulk added
+    documents = []
+
+    # Add the nodes to documents
+    for node in graph:
+        if not is_valid_node(node):
+            logger.warning(f'Invalid node encountered: {node}')
+        # Add a 'type' parameter to avoid confusing nodes and links
+        n = deepcopy(node)
+        n[TYPE] = NODE
+        documents.append(n)
+
+    # Add the edges to documents
+    edges = to_sbel(graph, yield_metadata=False)
+    for edge in edges:
+        if not is_valid_edge(edge):
+            logger.warning(f'Invalid edge encountered: {edge}')
+        # Add a 'type' parameter to avoid confusing nodes and links
+        e = deepcopy(edge)
+        e[TYPE] = LINK
+        documents.append(e)
+    # Insert the documents into the collection
+    collection.insert_many(documents)
+    return collection
+
+
+def find_nodes(
+    collection: Collection,
+    name: str = None,
+    identifier: str = None,
+    function: str = None,
+    variants: List[Variant] = None,
+) -> List[Mapping[str, Any]]:
+    """Find all the nodes that match the given criteria from a MongoDB Collection where a graph is stored.
+
+    :param collection: A MongoDB collection within a database where a PyBEL graph has been stored
+    :param name: The name of the desired node
+    :param identifier: The identifier of the desired node
+    :param function: The type of the desired node ("protein", "complex", etc)
+    :param variants: A list of variants that the desired node should contain.
+     Note: nodes that contain the variants in addition to specified variants will be matched.
+    :return: A list containing all the nodes that match the given criteria
+    """
+    if not (name or identifier):
```

Comment:
an identifier is pretty useless without a namespace, isn't it?

Reply:
Should I remove the identifier option or add in a namespace param?